### PR TITLE
Match Match func_800738A4 and WIP other ghost data funcs,

### PR DESCRIPTION
**Pull request by Ryan-Myers**
I think I'm going to give up on trying to figure out ghost data structs. I've added WIP functions for the functions using the data though. Also did some other minor cleanup. I tried to get rid of any (f64) casts with something that felt a little more plausible. It will also make it easier to remove doubles in the future with a compiler flag to treat any floats without an f as single precision as well.

### CODE REVIEW

**Review by DavidSM64 — APPROVED** [review]
LGTM

File: src/save_data.c [hunk 1]
```diff
@@ -671,10 +667,10 @@ SIDeviceStatus get_file_extension(s32 controllerIndex, s32 fileType, char *fileE
                 continue;
             }
             if (fileType == 3) {
-                if((func_800CE050((u8 *) fileNames[fileNum], (char *)D_800E7630, strlen((char *)D_800E7640)) != 0)) {
+                if((func_800CE050((u8 *) fileNames[fileNum], (char *)sDKRacingAdv1, strlen((char *)sDKRacingAdv2)) != 0)) {
```
Comment (FazanaJ) [inline src/save_data.c:670]:
```suggestion
                if((func_800CE050((u8 *) fileNames[fileNum], (char *) sDKRacingAdv1, strlen((char *) sDKRacingAdv2)) != 0)) {
```

File: src/save_data.c [hunk 2]
```diff
@@ -671,10 +667,10 @@ SIDeviceStatus get_file_extension(s32 controllerIndex, s32 fileType, char *fileE
...
                     continue;
                 }
-            } else if((func_800CE050((u8 *) fileNames[fileNum], (char *)D_800E7650, strlen((char *)D_800E7660)) != 0))  {
+            } else if((func_800CE050((u8 *) fileNames[fileNum], (char *)sDKRacingTimes1, strlen((char *)sDKRacingTimes2)) != 0))  {
```
Comment (FazanaJ) [inline src/save_data.c:673]:
```suggestion
            } else if((func_800CE050((u8 *) fileNames[fileNum], (char *) sDKRacingTimes1, strlen((char *) sDKRacingTimes2)) != 0))  {
```

File: src/save_data.c [hunk 3]
```diff
@@ -715,8 +711,7 @@ s32 read_game_data_from_controller_pak(s32 controllerIndex, char *fileExt, Setti
         start_reading_controller_data(controllerIndex);
         return (controllerIndex << 30) | ret;
     }
-    //D_800E7670 = "DKRACING-ADV"
-    ret = get_file_number(controllerIndex, (char *)D_800E7670, fileExt, &fileNumber);
+    ret = get_file_number(controllerIndex, (char *)sDKRacingAdv3, fileExt, &fileNumber);
```
Comment (FazanaJ) [inline src/save_data.c:714]:
```suggestion
    ret = get_file_number(controllerIndex, (char *) sDKRacingAdv3, fileExt, &fileNumber);
```

File: src/save_data.c [hunk 4]
```diff
@@ -762,8 +757,7 @@ s32 write_game_data_to_controller_pak(s32 controllerIndex, Settings *arg1) {
     func_800732E8(arg1, gameData + 4);
     ret = get_file_extension(controllerIndex, 3, (char *)&fileExt);
     if (ret == CONTROLLER_PAK_GOOD) {
-        // D_800E7680 = DKRACING-ADV
-        ret = write_controller_pak_file(controllerIndex, -1, (char *) D_800E7680, (char *)&fileExt, gameData, fileSize);
+        ret = write_controller_pak_file(controllerIndex, -1, (char *) sDKRacingAdv4, (char *)&fileExt, gameData, fileSize);
```
Comment (FazanaJ) [inline src/save_data.c:760]:
```suggestion
        ret = write_controller_pak_file(controllerIndex, -1, (char *) sDKRacingAdv4, (char *) &fileExt, gameData, fileSize);
```

File: src/save_data.c [hunk 5]
```diff
@@ -785,8 +779,7 @@ s32 read_time_data_from_controller_pak(s32 controllerIndex, char *fileExt, Setti
         return (controllerIndex << 30) | status;
     }
 
-    //D_800E7690 = DKRACING-TIMES
-    status = get_file_number(controllerIndex, (char *)D_800E7690, fileExt, &fileNumber);
+    status = get_file_number(controllerIndex, (char *)sDKRacingTimes3, fileExt, &fileNumber);
```
Comment (FazanaJ) [inline src/save_data.c:782]:
```suggestion
    status = get_file_number(controllerIndex, (char *) sDKRacingTimes3, fileExt, &fileNumber);
```

File: src/save_data.c [hunk 6]
```diff
@@ -830,8 +823,7 @@ s32 write_time_data_to_controller_pak(s32 controllerIndex, Settings *arg1) {
     func_800738A4(arg1, timeData + 4);
     ret = get_file_extension(controllerIndex, 4, (char *)&fileExt);
     if (ret == CONTROLLER_PAK_GOOD) {
-        //D_800E76A0 = DKRACING-TIMES
-        ret = write_controller_pak_file(controllerIndex, -1, (char *)D_800E76A0, (char *)&fileExt, timeData, fileSize);
+        ret = write_controller_pak_file(controllerIndex, -1, (char *)sDKRacingTimes4, (char *)&fileExt, timeData, fileSize);
```
Comment (FazanaJ) [inline src/save_data.c:826]:
```suggestion
        ret = write_controller_pak_file(controllerIndex, -1, (char *) sDKRacingTimes4, (char *) &fileExt, timeData, fileSize);
```

**Review by FazanaJ — COMMENTED** [review]
funny pedantic styling otherwise all good

**Pull request merged**